nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/test/java/org/apache/nifi/processors/kudu/TestScanKudu.java

```diff
@@ -0,0 +1,449 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.kudu;
+
+import org.apache.kudu.client.KuduException;
+import org.apache.kudu.test.KuduTestHarness;
+import org.apache.kudu.test.cluster.MiniKuduCluster;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class TestScanKudu {
```

Comment:
The current tests only work with String columns so I wrote one that covers (mostly) all the currently supported types.
Sure enough, there are some small bugs in handling some of the other types.

Here's the unit test (will comment the bugs in the corresponding class):
```java
    @Test
    public void testScanKuduWithMultipleTypes() throws KuduException {
        List<ColumnSchema> columns = Arrays.asList(
                new ColumnSchema.ColumnSchemaBuilder("key", Type.INT8).key(true).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.INT16.getName(), Type.INT16).key(true).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.INT32.getName(), Type.INT32).key(true).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.INT64.getName(), Type.INT64).key(true).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.BINARY.getName(), Type.BINARY).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.STRING.getName(), Type.STRING).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.BOOL.getName(), Type.BOOL).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.FLOAT.getName(), Type.FLOAT).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.DOUBLE.getName(), Type.DOUBLE).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.UNIXTIME_MICROS.getName(), Type.UNIXTIME_MICROS).key(false).build(),
                new ColumnSchema.ColumnSchemaBuilder(Type.DECIMAL.getName(), Type.DECIMAL).typeAttributes(
                        new ColumnTypeAttributes.ColumnTypeAttributesBuilder()
                                .precision(20)
                                .scale(4)
                                .build()
                ).key(false).build()
        );

        Instant now = Instant.now();

        KuduTable kuduTable = kuduScan.getKuduTable(DEFAULT_TABLE_NAME, columns);
        Insert insert = kuduTable.newInsert();
        PartialRow rows = insert.getRow();
        rows.addByte("key", (byte) 1);
        rows.addShort(Type.INT16.getName(), (short)20);
        rows.addInt(Type.INT32.getName(), 300);
        rows.addLong(Type.INT64.getName(), 4000L);
        rows.addBinary(Type.BINARY.getName(), new byte[]{55, 89});
        rows.addString(Type.STRING.getName(), "stringValue");
        rows.addBoolean(Type.BOOL.getName(), true);
        rows.addFloat(Type.FLOAT.getName(), 1.5F);
        rows.addDouble(Type.DOUBLE.getName(), 10.28);
        rows.addTimestamp(Type.UNIXTIME_MICROS.getName(), Timestamp.from(now));
        rows.addDecimal(Type.DECIMAL.getName(), new BigDecimal("3.1415"));

        KuduSession kuduSession = kuduScan.kuduClient.newSession();
        kuduSession.apply(insert);
        kuduSession.close();

        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);

        runner.setIncomingConnection(false);
        runner.enqueue();
        runner.run(1, false);

        runner.assertTransferCount(ScanKudu.REL_FAILURE, 0);
        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 1);
        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 1);

        final MockFlowFile flowFile = runner.getFlowFilesForRelationship(ScanKudu.REL_SUCCESS).get(0);
        Object timestamp = ChronoUnit.MICROS.between(Instant.EPOCH, now);
        flowFile.assertContentEquals("[{\"rows\":[{\"key\":\"1\",\"int16\":\"20\",\"int32\":\"300\",\"int64\":\"4000\",\"binary\":\"0x3759\",\"string\":\"stringValue\",\"bool\":\"true\",\"float\":\"1.5\",\"double\":\"10.28\",\"unixtime_micros\":\"" + timestamp + "\",\"decimal\":\"3.1415\"}]}]");
    }
```